src/parsing.rs

```diff
@@ -281,7 +290,7 @@ fn parse_type(types: &mut IntermediateTypes, parent_visitor: &ParentVisitor, typ
                     match control {
                         ControlOperator::Range(min_max) => {
                             match cddl_ident.to_string().as_str() {
-                                "int" | "uint" => match range_to_primitive(min_max.0, min_max.1) {
+                                "int" | "uint" | "float" => match range_to_primitive(min_max.0, min_max.1) {
```

Comment:
What if someone uses one of the concrete types (`float32`, `float64`) here as the identifier?